crates/ty_python_semantic/resources/mdtest/override.md

```diff
@@ -0,0 +1,314 @@
+    def eggs(): ...  # error: [explicit-override]
+    @property
+    @override
+    def bad_property1(self) -> int: ...  # TODO: should emit `invalid-override` here
+    @override
+    @property
+    def bad_property2(self) -> int: ...  # TODO: should emit `invalid-override` here
+    @lossy_decorator
+    @override
+    def lossy(self): ...  # TODO: should emit `invalid-override` here
+    @override
+    @lossy_decorator
+    def lossy2(self): ...  # TODO: should emit `invalid-override` here
+
+# TODO: all overrides in this class should cause us to emit *Liskov* violations,
+# but not `@override` violations
+class LiskovViolatingButNotOverrideViolating(Parent):
+    @override
+    @property
+    def foo(self) -> int: ...
+    @override
+    def my_property1(self) -> int: ...
+    @staticmethod
+    @override
+    def class_method1() -> int: ...
+    @classmethod
+    @override
+    def static_method1(cls) -> int: ...
+
+# Diagnostic edge case: `override` is very far away from the method definition in the source code:
+
+T = TypeVar("T")
+
+def identity(x: T) -> T: ...
+
+class Foo:
+    @override
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    @identity
+    def bar(self): ...  # error: [explicit-override]
+```
+
+## Overloads
+
+The typing spec states that for an overloaded method, `@override` should only be applied to the
+implementation function. However, we nonetheless respect the decorator in this situation, even =
+though we also emit `invalid-overload` on these methods.
+
+```py
+from typing_extensions import override, overload
+
+class Spam:
+    @overload
+    def foo(self, x: str) -> str: ...
+    @overload
+    def foo(self, x: int) -> int: ...
+    @override
+    def foo(self, x: str | int) -> str | int:  # error: [explicit-override]
+        return x
+
+    @overload
+    @override
+    def bar(self, x: str) -> str: ...
+    @overload
+    @override
+    def bar(self, x: int) -> int: ...
+    @override
+    # error: [invalid-overload] "`@override` decorator should be applied only to the overload implementation"
+    # error: [invalid-overload] "`@override` decorator should be applied only to the overload implementation"
+    # error: [explicit-override]
+    def bar(self, x: str | int) -> str | int:
+        return x
+
+    @overload
+    @override
+    def baz(self, x: str) -> str: ...
+    @overload
+    def baz(self, x: int) -> int: ...
+    # error: [invalid-overload] "`@override` decorator should be applied only to the overload implementation"
+    # error: [explicit-override]
+    def baz(self, x: str | int) -> str | int:
+        return x
+```
+
+In a stub file, `@override` should always be applied to the first overload. Even if it isn't, we
+always emit `explicit-override` diagnostics on the first overload.
```

Comment:
It's a bit annoying to implement this but it means that we pass several typing conformance-suite tests that we'd otherwise fail, so I think it's worth hit